integration/autoupdate/e2e/main.go

```diff
@@ -0,0 +1,237 @@
+//go:build e2e
+// +build e2e
+
+/*
+ * Teleport
+ * Copyright (C) 2025  Gravitational, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package main
+
+import (
+	"bufio"
+	"context"
+	"crypto/sha256"
+	"encoding/hex"
+	"fmt"
+	"io"
+	"log/slog"
+	"net/http"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"runtime"
+	"strings"
+
+	"github.com/gravitational/teleport/integration/helpers/archive"
+)
+
+// Source is used for defining Teleport source directory by local ID.
+type Source struct {
+	ID   int
+	Path string
+}
+
+// VersionInfo defines what version label must be used during compilation.
+type VersionInfo struct {
+	Version  string
+	SourceID int
+}
+
+var sources = []Source{
+	// Path to Teleport source code with previous v1 version of CTMU.
+	{ID: 1, Path: "../teleport-sync"},
+	// Path to Teleport source code with latest version of CTMU.
+	{ID: 2, Path: "./"},
+}
+
+var versions = []VersionInfo{
+	{"17.5.1", 1},
+	{"17.5.4", 1},
+	{"17.5.7", 1},
+	{"18.0.0", 2},
+	{"18.5.5", 2},
+	{"18.5.7", 2},
+}
+
+func main() {
+	ctx := context.Background()
+	outputDir := "./test-packages"
+	if err := os.MkdirAll(outputDir, 0755); err != nil {
+		slog.ErrorContext(ctx, "Failed to create output directory: %v", err)
```

Comment:
Is there a reason for using slog with Printf style, rather than:

```go
slog.ErrorContext(ctx, "Failed to create output directory", "error", err)
```

The same applies to three more slog.ErrorContext calls below.